cpp/src/Ice/ProtocolInstance.cpp

```diff
@@ -25,9 +25,9 @@ IceInternal::ProtocolInstance::ProtocolInstance(
     const string& protocol,
     bool secure)
     : _instance(getInstance(com)),
-      _traceLevel(_instance->traceLevels()->network),
-      _traceCategory(_instance->traceLevels()->networkCat),
-      _properties(_instance->initializationData().properties),
+      _traceLevel(_instance.lock()->traceLevels()->network),
```

Comment:
This shouldn't be a problem the `com` parameter has a strong reference to the instance.